File: qc-core/src/main/java/com/eager/questioncloud/core/domain/post/implement/PostCommentRemover.java
package com.eager.questioncloud.core.domain.post.implement;

import com.eager.questioncloud.core.domain.post.model.PostComment;
import com.eager.questioncloud.core.domain.post.repository.PostCommentRepository;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

@Component
@RequiredArgsConstructor
public class PostCommentRemover {
    private final PostCommentRepository postCommentRepository;

    public void delete(Long commentId, Long userId) {
        PostComment postComment = postCommentRepository.findByIdAndWriterId(commentId, userId);
        postCommentRepository.delete(postComment);
    }
}

File: qc-core/qc-post/qc-post-api/src/main/kotlin/com/eager/questioncloud/post/implement/PostCommentRemover.kt
package com.eager.questioncloud.post.implement

import com.eager.questioncloud.post.command.DeletePostCommentCommand
import com.eager.questioncloud.post.repository.PostCommentRepository
import org.springframework.stereotype.Component

@Component
class PostCommentRemover(
    private val postCommentRepository: PostCommentRepository
) {
    fun deletePostComment(command: DeletePostCommentCommand) {
        val postComment = postCommentRepository.findByIdAndWriterId(command.commentId, command.userId)
        postCommentRepository.delete(postComment)
    }
}